presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/RpmController.java

```diff
@@ -93,46 +91,15 @@ public void uninstallAsync(boolean checkDependencies)
         LOGGER.debug("Successfully uninstalled Presto");
     }
 
-    public void upgradeAsync(URL packageUrl, boolean checkDependencies, boolean preserveConfig)
-            throws PrestoManagerException
-    {
-        File tempPackage = getRpmPackage(packageUrl);
-        try {
-            if (preserveConfig) {
-                File tempConfig = storeConfigFiles(CONFIG_DIR);
-                try {
-                    upgradePackage(tempPackage.toString(), checkDependencies);
-                    deployConfigFiles(tempConfig, CONFIG_DIR);
-                }
-                finally {
-                    if (!tempConfig.delete()) {
-                        LOGGER.warn("Failed to delete the temporary file: %s", tempConfig.toString());
-                    }
-                }
-            }
-            else {
-                upgradePackage(tempPackage.toString(), checkDependencies);
-                addConfigFiles(CONFIG_DIR);
-                addConnectors(CATALOG_DIR);
-            }
-        }
-        finally {
-            if (!tempPackage.delete()) {
-                LOGGER.warn("Failed to delete the temporary file: %s", tempPackage.toString());
-            }
-        }
-        LOGGER.debug("Successfully upgraded presto");
-    }
-
-    private void upgradePackage(String pathToRpm, boolean checkDependencies)
+    public void upgradePackage(File packageFile, boolean checkDependencies)
             throws PrestoManagerException
     {
         int upgradeRpm;
         if (checkDependencies) {
-            upgradeRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-U", pathToRpm);
+            upgradeRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-U", packageFile.toString());
```

Comment:
This isn't so important, but I keep forgetting to mention it whenever this file changes: it's better to use the long argument names for command line tools when they exist (e.g. `--upgrade` vs `-U`).